pallets/liquid-staking-two/src/lib.rs

```diff
@@ -0,0 +1,820 @@
+// Copyright 2021 Parallel Finance Developer.
+// This file is part of Parallel Finance.
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// http://www.apache.org/licenses/LICENSE-2.0
+
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//! # Liquid staking pallet
+//!
+//! ## Overview
+//!
+//! This pallet manages the NPoS operations for relay chain asset.
+
+// 1. update parachain era before relaychain era updated, a few blocks in advance,
+// 2. calculate parachain status, whether bond/unbond/rebond,
+// 3. invoke relaychain staking methods, bond/unbond/rebond,
+// 4. invoke parachain method record relaychain resonse, whether bond/unbond successed or failed,
+// 5. waiting relaychain era updated, get the reward amount.
+// 6. record in parachain with blocknumber when real relaychain era updated.
+// 7. record reward on parachain. update parachain exchange rate.
+// 8. if step-4 successed, mint/deposit xToken out according to current exchangerate(after record reward),
+#![cfg_attr(not(feature = "std"), no_std)]
+
+use frame_support::{
+    pallet_prelude::*, traits::SortedMembers, transactional, BoundedVec, PalletId,
+};
+use frame_system::pallet_prelude::*;
+use orml_traits::XcmTransfer;
+use sp_runtime::{traits::AccountIdConversion, ArithmeticError, FixedPointNumber, RuntimeDebug};
+
+use sp_std::prelude::*;
+use xcm::v0::{Junction, MultiLocation, NetworkId};
+
+use orml_traits::{MultiCurrency, MultiCurrencyExtended};
+
+pub use pallet::*;
+use primitives::{
+    Amount, Balance, CurrencyId, EraIndex, ExchangeRateProvider,
+    LiquidStakingProtocol, Rate, Ratio,
+};
+
+pub const MAX_UNSTAKE_CHUNKS: usize = 5;
+
+#[derive(Copy, Clone, Eq, PartialEq, Encode, Decode, RuntimeDebug)]
+pub enum StakingOperationType {
+    Bond,
+    Unbond,
+    Rebond,
+    Matching,
+    TransferToRelaychain,
+    RecordReward,
+    RecordSlash,
+}
+
+#[derive(Copy, Clone, Eq, PartialEq, Encode, Decode, RuntimeDebug)]
+pub enum ResponseStatus {
```

Comment:
can you give some comments on the meaning of every status and how stake-client should use it?